timeseries/tests/unittests/models/test_models.py

```diff
@@ -26,649 +23,479 @@
-def test_given_hyperparameter_spaces_to_init_when_fit_called_then_error_is_raised(model_class, temp_model_path):
-    model = model_class(
-        path=temp_model_path,
-        freq="h",
-        quantile_levels=[0.1, 0.9],
-        hyperparameters={
-            "max_epochs": space.Int(3, 4),
-        },
-    )
-    with pytest.raises(ValueError, match=".*hyperparameter_tune.*"):
-        model.fit(
+        hpo_results, _ = model.hyperparameter_tune(
+            hyperparameter_tune_kwargs={"num_trials": num_trials, "scheduler": "local", "searcher": "random"},
+            time_limit=300,
             train_data=DUMMY_TS_DATAFRAME,
+            val_data=val_data,
         )
+        assert len(hpo_results) == num_trials
+        for result in hpo_results.values():
+            assert 1 <= result["hyperparameters"]["max_epochs"] <= 3
+            assert np.isfinite(result["val_score"])
 
+    def test_when_hyperparameter_spaces_provided_to_init_and_fit_called_then_error_is_raised(
+        self, model_class, temp_model_path
+    ):
+        model = model_class(
+            path=temp_model_path,
+            freq="h",
+            quantile_levels=[0.1, 0.9],
+            hyperparameters={
+                "max_epochs": space.Int(3, 4),
+            },
+        )
+        with pytest.raises(ValueError, match=".*hyperparameter_tune.*"):
+            model.fit(
+                train_data=DUMMY_TS_DATAFRAME,
+            )
 
-@pytest.mark.parametrize("model_class", TESTABLE_MODELS)
-@pytest.mark.parametrize(
-    "quantile_levels",
-    [
-        [0.1, 0.44, 0.72],
-        [0.1, 0.5, 0.9],
-    ],
-)
-def test_when_fit_called_then_models_train_and_returned_predictor_inference_has_mean_and_correct_quantiles(
-    model_class, quantile_levels, temp_model_path, dummy_hyperparameters
-):
-    model = model_class(
-        path=temp_model_path,
-        freq="h",
-        prediction_length=3,
-        quantile_levels=quantile_levels,
-        hyperparameters=dummy_hyperparameters,
-    )
-    model.fit(train_data=DUMMY_TS_DATAFRAME)
-    predictions = model.predict(DUMMY_TS_DATAFRAME, quantile_levels=quantile_levels)
-
-    assert isinstance(predictions, TimeSeriesDataFrame)
 
-    predicted_item_index = predictions.item_ids
-    expected_columns = ["mean"] + [str(q) for q in quantile_levels]
-    assert all(predicted_item_index == DUMMY_TS_DATAFRAME.item_ids)  # noqa
-    assert (predictions.columns == expected_columns).all()
+class TestAllModelsWhenCustomProblemSpecificationsProvided:
+    """Test all models with varying forecast problem specifications such as frequency of the
+    time series, quantile levels, item index, etc.
+    """
 
+    @pytest.mark.parametrize(
+        "quantile_levels",
+        [
+            [0.1, 0.44, 0.72],
+            [0.1, 0.5, 0.9],
+        ],
+    )
+    def test_when_fit_called_then_models_train_and_returned_predictor_inference_has_mean_and_correct_quantiles(
```

Comment:
Again not sure about `predictor_inference` here. Maybe `predictions` or `forecast`?